packages/ods/src/components/modal/src/components/ods-modal/ods-modal.tsx

```diff
@@ -0,0 +1,127 @@
+import { Component, Element, Event, type EventEmitter, type FunctionalComponent, Host, Listen, Method, Prop, Watch, h } from '@stencil/core';
+import { ODS_ICON_NAME } from '../../../../icon/src';
+import { ODS_MODAL_COLOR, type OdsModalColor } from '../../constants/modal-color';
+
+@Component({
+  shadow: true,
+  styleUrl: 'ods-modal.scss',
+  tag: 'ods-modal',
+})
+export class OdsModal {
+  private modalDialog!: HTMLDialogElement;
+
+  @Element() el!: HTMLElement;
+
+  @Prop({ reflect: true }) public color: OdsModalColor = ODS_MODAL_COLOR.information;
+  @Prop({ reflect: true }) public isDismissible: boolean = true;
+  @Prop({ mutable: true, reflect: true }) public isOpen: boolean = false;
+
+  @Event() odsModalClose!: EventEmitter<void>;
+  @Event() odsModalOpen!: EventEmitter<void>;
+
+  @Method()
+  async close(): Promise<void> {
+    document.body.style.removeProperty('overflow');
+
+    if (this.modalDialog) {
+      this.modalDialog.classList.add('ods-modal__dialog--close-animation');
+
+      const onAnimationEnd = (): void => {
+        this.isOpen = false;
+        this.modalDialog.classList.remove('ods-modal__dialog--close-animation');
+        this.modalDialog.removeEventListener('animationend', onAnimationEnd);
+      };
+
+      this.modalDialog.addEventListener('animationend', onAnimationEnd);
+    } else {
+      this.isOpen = false;
+    }
+
+    this.odsModalClose.emit();
+  }
+
+  @Method()
+  async open(): Promise<void> {
+    document.body.style.overflow = 'hidden';
```

Comment:
there is still an issue when the web-component is rendered with `isOpen` as it does not go through this.

Instead of using JS, maybe be safer to use css, try something like:
```
body:has(dialog[open]) {
  overflow: hidden;
}
```